junebug/api.py

```diff
@@ -338,5 +356,70 @@ def create_router(self, request, body):
         ))
 
     @app.route('/health', methods=['GET'])
+    @inlineCallbacks
     def health_status(self, request):
-        return response(request, 'health ok', {})
+        if self.config.rabbitmq_management_interface:
+
+            ids = yield Channel.get_all(self.redis)
+
+            def get_queues(channel_ids):
+                gets = []
+                for channel_id in channel_ids:
+                    for sub in ['inbound', 'outbound', 'event']:
+
+                        queue_name = "%s.%s" % (channel_id, sub)
+                        url = 'http://%s/api/queues/%s/%s' % (
+                            self.config.rabbitmq_management_interface,
+                            self.amqp_config['vhost'].replace('/', '%2F'),
+                            queue_name
+                        )
+                        http_client = HTTPClient(
+                            self.agent_factory(
+                                self.clock,
+                                pool=self.pool_factory(self.clock)))
+
+                        get = http_client.get(
+                            url, auth=(self.amqp_config['username'],
+                                       self.amqp_config['password']))
```

Comment:
This is all quite messy having these things nested so deep. Could we put all the making the url and HTTP request things into a rabbitmq management client class, so that you just go `rabbitmq_management_client.get_queue(vhost, queue_name)`